Problem2.cpp

```diff
@@ -0,0 +1,56 @@
+#include<iostream>
+using namespace std;
+
+
+
+
+
+int binarySearch(int arr[], int left, int right)    //Binary search implemented.
+{
+    int min = INT_MAX;
+    int mid = left + (right - left) / 2;
+    if (left<=right) {
+        if(left==right)         //if only one element in the array. CASE 1.
+        {
+            cout<<left; return left;
+        }
+
+        
+        if(arr[mid]>arr[mid-1] && arr[mid]<arr[mid+1])          //CASE 2: if middle is > left and smaller than right. Search the right part. 
```

Comment:
this is not correct as this way you are always thinking that the mid is always the point where the array is sorted which is not always the case